purchase_delivery_address/model/purchase_order.py

```diff
@@ -15,34 +15,59 @@
 #    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from openerp import models, api
+from openerp import models, api, exceptions
 
 
 class PurchaseOrder(models.Model):
     _inherit = 'purchase.order'
 
-    @api.multi
-    def onchange_dest_address_id(self, dest_address_id):
-        """If we enter a delivery address, normally it is a dropshipping-like
-        situation, so we choose an appropriate picking type.
+    @api.onchange('dest_address_id')
+    def new_onchange_dest_address_id(self):
+        """Find a picking type from the address
+
+        If the address can be an internal warehouse or a customer, the picking
+        type is changed accordingly. This intentionally overrides the original
+        without super, and should be consistent with the module
+        purchase_requisition_delivery_address.
```

Comment:
@pedrobaeza I tried at the time, but then I went into some strange issues, probably because of interactions between old-style and new-style onchanges.